### ISSUE TITLE
Implement folding for `.tree` files

### ISSUE THREAD

**Issue by PraneshASP** [PraneshASP/vscode-solidity-inspector#37]
Ref: https://x.com/lucasmanuel_eth/status/1737466723568128394

**Comment by PaulRBerg** [comment]
I've just bumped into this - I wanted to be able to fold this subsection of our [withdraw tree](https://github.com/sablier-labs/v2-core/blob/e080f20eafef0fc18049bcc77f1694db043860f1/test/integration/concrete/lockup/withdraw/withdraw.tree#L20) but I couldn't do it:

<img width="400" alt="SCR-20240329-lvdu" src="https://github.com/PraneshASP/vscode-solidity-inspector/assets/8782666/13f6a6f8-9451-477e-b137-2a76127b7ca6">

**Comment by PraneshASP** [comment]
yeah I was facing some issues as well. VSCode provides inbuilt support for folding codeblocks, but it doesn't work well for tree files. I started looking into it a while back, but didn't make much progress as it got quite complex than I anticipated. Will get back to it in a week or 2.
